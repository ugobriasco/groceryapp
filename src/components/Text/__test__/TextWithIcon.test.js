import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import Icon from  'react-native-vector-icons/MaterialIcons';

import {TextWithIcon, styles} from '../';

describe('TextWithIcon', () => {

	it('renders correctly', () => {
	const tree = renderer.create(<TextWithIcon />).toJSON();
	expect(tree).toBeTruthy();

	});

	it ('styles an object', () => {
		expect(typeof styles).toBe('object');
	});

	describe('size styling props', () => {
		it('should render a given tex wit customized color, background and icon', () => {
			const rendered = renderer.create(
				<TextWithIcon 
					text="foo" 
					backgroundColor="red"
					textColor="blue" 
					iconName="add"
				/>);
			expect(rendered).toMatchSnapshot();	
		});
	});
});



