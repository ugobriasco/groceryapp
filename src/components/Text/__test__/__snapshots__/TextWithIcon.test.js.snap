// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithIcon size styling props should render a given tex wit customized color, background and icon 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "blue",
          "fontSize": 40,
        },
        Array [
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î……
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        undefined,
        Object {
          "color": "blue",
        },
      ]
    }
  >
    foo
  </Text>
</View>
`;
